### ISSUE TITLE
Support `gpt-4o-2024-08-06` model

### ISSUE THREAD

**Issue by MatanYadaev** [promptfoo/promptfoo#1402]
When I try to run OpenAI latest model I get this error:
```
Error: Unknown OpenAI model type: gpt-4o-2024-08-06. Use one of the following providers: openai:chat:<model name>, openai:completion:<model name>, openai:embeddings:<model name>, openai:image:<model name>
```

The issue occurs because the latest model wasn't mapped yet in promptfoo source code. You can see it here: https://github.com/promptfoo/promptfoo/blob/main/src/providers/openai.ts#L50-L56

I would suggest to not throw an error when a model isn't found. It blocks us, the clients of this package, from using latest models until promptfoo authors update the mapping. A warning might be better.

**Comment by typpo** [comment]
I assume you are trying `openai:gpt-4o-2024-08-06`?  You should be able to use `openai:chat:gpt-4o-2024-08-06` without waiting for an update.  

The reason this is necessary is that it wouldn't know whether to use the `chat` or `completion` API.  Though nowadays it's a pretty safe bet that an unknown model is a chat model.

**Comment by MatanYadaev** [comment]
@typpo thanks for the quick answer!

**Comment by typpo** [comment]
Landed in 0.76.0
